x-pack/plugins/security_solution/public/detections/components/rules/eql_query_bar/footer.tsx

```diff
@@ -28,22 +49,169 @@ const FlexGroup = styled(EuiFlexGroup)`
   min-height: ${({ theme }) => theme.eui.euiSizeXL};
 `;
 
+const FlexItemLeftBorder = styled(EuiFlexItem)`
+  border-left: 1px solid ${({ theme }) => theme.eui.euiColorLightShade};
+`;
+
+const FlexItemWithMarginRight = styled(EuiFlexItem)`
+  margin-right: ${({ theme }) => theme.eui.euiSizeS};
+`;
+
 const Spinner = styled(EuiLoadingSpinner)`
   margin: 0 ${({ theme }) => theme.eui.euiSizeS};
 `;
 
-export const EqlQueryBarFooter: FC<Props> = ({ errors, isLoading }) => (
-  <Container>
-    <FlexGroup alignItems="center" justifyContent="spaceBetween" gutterSize="none">
-      <EuiFlexItem>
-        {errors.length > 0 && (
-          <ErrorsPopover ariaLabel={i18n.EQL_VALIDATION_ERROR_POPOVER_LABEL} errors={errors} />
+const singleSelection = { asPlainText: true };
+
+export const EqlQueryBarFooter: FC<Props> = ({
+  errors,
+  isLoading,
+  optionsData,
+  optionsSelected,
+  onSelectLanguage,
+  onOptionsChange,
+}) => {
+  const [openEqlSettings, setIsOpenEqlSettings] = useState(false);
+
+  const openEqlSettingsHandle = useCallback(() => setIsOpenEqlSettings(true), []);
```

Comment:
Consider renaming `openEqlSettingsHandle` and `closeEqlSettingsHandle` to `openEqlSettingsHandler` and `closeEqlSettingsHandler` respectively, because singular `Handle` often describes a "thing", like a file handle, vs the plural Handler, which typically describes "a callback".